doc/installing.rst

```diff
@@ -24,7 +24,7 @@ The package and version requirements for ``PypeIt`` are:
 * `scipy <http://www.scipy.org/>`_ version 1.4 or later
 * `matplotlib <http://matplotlib.org/>`_  version 3.1 or later
 * `numba <https://numba.pydata.org/>`_ version 0.39.0 or later
-* `PySide2 <https://wiki.qt.io/Qt_for_Python>`_ version 5
+* `QtPy <https://pypi.org/project/qtpy>`_ version 1.9 or later
```

Comment:
I'm not sure I follow how the installation works now in the sense that QtPy is a abstracting wrapper for the other packages.  Does pip install qtpy install one of these packages if none are installed on the system?  We might want to explain how this works in the installation docs.  The QtPy PyPI page says "You need PyQt5, PyQt4, PySide2 or PySide installed in your system to make use of QtPy.", and we might want to repeat something like this in the install doc.

Reply:
That's a good point! I was thinking that there are generally 2 types of environments that users install PypeIt into:

1. a full environment with a whole bunch of stuff in it, some of which will probably require PyQt5 or PySide2, so the user will already have a required python Qt binding installed and ready to use
2. an empty environment in which case guis won't work. In this case we definitely need to provide guidance to users on what to do, but I think presenting users with a full list of requirements before quick and easy install instructions can be a little intimidating, so I think I'll change that when I do a dive into the install docs later (but still in this PR).

I also think its valuable to have a one-line install command presented at the top of the install docs for non-developer users who want PypeIt to Just Work. My goal with creating and maintaining the conda-forge recipe for PypeIt is to provide users with this experience, but I do know that there are varying opinions within the PypeIt team on conda as a dependency manager/all-in-one install tool, so I don't want to put forward `conda install -c conda-forge pypeit` as the recommended way to install PypeIt without greater buy-in, but that that might be a conversation for the Slack.

Reply:
as a long-time conda advocate, my opinion is that pure python packages like `pypeit` should use `pip` as the recommended installation method. it's fine to provide advice on how to use conda and provide some recipes, but i know from lived experience that pulling in the `conda-forge` channel *will* cause problems for many users. 

what i am working on setting up is the following:

- `pip install pypeit` will install the core dependencies, but not any of the gui tools or dependencies
- `pip install pypeit[gui]` will install the gui dependencies (`ginga`, `linetools`, and `qtpy`) along with the scripts that use them
- `pip install pypeit[gui,pyside2]` or `pip install [gui,pyqt5]` will be provided as a convenience for pulling in the user's choice of QT binding if they don't already have it.

this would give the user more control over what they pull in based on their needs. iirc, conda doesn't have the same provision for such optional dependencies.